packages/table/src/cell/formats/truncatedFormat.tsx

```diff
@@ -80,6 +79,37 @@ export interface ITruncatedFormatProps extends IProps {
      * @default "..."
      */
     truncationSuffix?: string;
+
+    /**
+     * Approximate character width, used to determine whether to display the popover in approx truncation mode.
+     * The default value should work for normal table styles,
+     * but should be changed as necessary if the fonts or styles are changed significantly.
+     * @default 8
+     */
+    approximateCharWidth?: number;
+
+    /**
+     * Approximate character width, used to determine whether to display the popover in approx truncation mode.
+     * The default value should work for normal table styles, but should be changed if the fonts or styles are changed significantly.
+     * @default 18
+     */
+    approximateLineHeight?: number;
+
+    /**
+     * Total horizonal cell padding (both sides), used to determine whether to display the popover in approx truncation mode.
+     * The default value should work for normal table styles,
+     * but should be changed as necessary if the fonts or styles are changed significantly.
+     * @default 20
+     */
+    cellHorizontalPadding?: number;
+
+    /**
+     * Number of buffer lines desired, used to determine whether to display the popover in approx truncation mode.
```

Comment:
Hmm it's unclear to me what a buffer line is from reading these docs.

Also I wonder if these props should all be grouped in an object like `styleLayout: { approximateCharWidth, approximateLineHeight, ...}` because you're likely to set them all at once or not at all.

Reply:
Could certainly do this! The main issue with doing so is I'd have to merge the default props myself, since defaultProps are only shallowly merged.

Reply:
honestly we can get away with _not_ merging those default props for now. make the `styleLayout` object optional but all its fields required -- if this ends up feeling like a clunky API to actual users, we can reconsider, but I see no reason to do the extra work right now.